pkg/controller/cluster/controller.go

```diff
@@ -55,6 +61,7 @@ func (c *ClusterControllerImpl) Init(arguments sharedinformers.ControllerInitArg
 	c.kubernetesClientSet = arguments.GetSharedInformers().KubernetesClientSet
 
 	c.actuator = actuator
+	c.queue = arguments.GetSharedInformers().WorkerQueues["Cluster"]
```

Comment:
Are your pods restarting when testing this @detiber ? Just noticing I may have slipped a bug in with my PR, it seems to be crashing in the enqueueAfter method because this c.queue is nil, I think it's something to do with the order things are initialized. It was based on some code which originally fetched this WorkerQueue in the actual enqueueAfter method itself, I moved it as a result of comments from internal review, but I think that change might be bad and have slipped by as the pod crashes, restarts, and effectively re-updates status, so I didn't see it until now.